src/main/drivers/bus_i2c_hal.c

```diff
@@ -0,0 +1,324 @@
+/*
```

Comment:
I2C driver will have to be refactored to the same state-machine structure as current F1/F3/F4 i2c drivers. But this is a lower priority